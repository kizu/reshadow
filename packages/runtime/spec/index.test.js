/**
 * @jest-environment enzyme
 */

const getStyles = () =>
    [...document.getElementsByTagName('style')]
        .map(x => x.textContent)
        .join('');

let keyframes;
let css;

describe('runtime', () => {
    beforeEach(() => {
        [...document.getElementsByTagName('style')].forEach(x => {
            x.remove();
        });

        jest.isolateModules(() => {
            ({keyframes, css} = require('..'));
        });
    });

    it('should work with keyframes', () => {
        const pulse = keyframes`
            0%, 25% {
                opacity: 0;
            }
            100% {
                opacity: 1;
            }
        `;

        expect(pulse).toMatchSnapshot();
        expect(getStyles()).toMatchSnapshot();
    });

    it('should use dynamic variables fallback', () => {
        css.setOptions({variablesFallback: true});

        css`
            button {
                color: ${'red'};
            }
        `;

        expect(getStyles()).toMatchSnapshot();
    });
});
