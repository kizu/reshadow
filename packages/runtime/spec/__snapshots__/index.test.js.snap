// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtime should use dynamic variables fallback 1`] = `".__button_812yns{padding:2rem 4rem;color:red;border:0.5rem solid;}"`;

exports[`runtime should work with keyframes 1`] = `"_qrjiv5"`;

exports[`runtime should work with keyframes 2`] = `"@keyframes _qrjiv5{0%,25%{opacity:0;}100%{opacity:1;}}"`;
